src/main/java/com/atlassian/tutorial/macro/colorPick.java

```diff
@@ -0,0 +1,48 @@
+package com.atlassian.tutorial.macro;
+
+import com.atlassian.confluence.content.render.xhtml.ConversionContext;
+import com.atlassian.confluence.core.ContentEntityObject;
+import com.atlassian.confluence.macro.Macro;
+import com.atlassian.confluence.macro.MacroExecutionException;
+import com.atlassian.confluence.renderer.radeox.macros.MacroUtils;
+import com.atlassian.confluence.util.velocity.VelocityUtils;
+import com.atlassian.plugin.spring.scanner.annotation.component.Scanned;
+import com.atlassian.plugin.spring.scanner.annotation.imports.ComponentImport;
+import com.atlassian.webresource.api.assembler.PageBuilderService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Map;
+
+import static com.atlassian.confluence.user.AuthenticatedUserThreadLocal.getUsername;
+
+@Scanned
+public class colorPick implements Macro {
+
+    private PageBuilderService pageBuilderService;
+
+    @Autowired
+    public colorPick(@ComponentImport PageBuilderService pageBuilderService) {
+        this.pageBuilderService = pageBuilderService;
+    }
+
+    public String execute(Map<String, String> map, String s, ConversionContext conversionContext) throws MacroExecutionException {
```

Comment:
add final for all parameters

Reply:
what is ``s`` and why is it needed here?

Reply:
`s` contains body of macro. I don't use this variable, but i can't remove it because `execute` method implemented from `Macro` interface with it